packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostUtils.js

```diff
@@ -0,0 +1,578 @@
+ */
+async function initTextToSpeech() {
+  // Ensure services get initialized only once per session.
+  if (isTextToSpeechInitialized) return;
+  isTextToSpeechInitialized = true;
+
+  // Enable Polly service functionality.
+  const polly = new AWS.Polly();
+  const presigner = new AWS.Polly.Presigner();
+  await aws.TextToSpeechFeature.initializeService(polly, presigner, AWS.VERSION);
+}
+
+/**
+ * Loads the assets that comprise a host character.
+ *
+ * @private
+ *
+ * @param {*} scene
+ * @param {*} characterConfig See the createHost() function docs above.
+ * @return {Object} An object containing the loaded assets organized as follows:
+ ```
+{
+  characterMesh: BABYLON.Mesh,
+  animClips: {
+    idleClips: BABYLON.AnimationGroup[],
+    lipSyncClips: BABYLON.AnimationGroup[],
+    gestureClips: BABYLON.AnimationGroup[],
+    emoteClips: BABYLON.AnimationGroup[],
+    faceClips: BABYLON.AnimationGroup[],
+    blinkClips: BABYLON.AnimationGroup[],
+    poiClips: BABYLON.AnimationGroup[],
+  }
+  bindPoseOffset: BABYLON.AnimationGroup,
+  gestureConfig: Object,  // see "3d-assets/animations/adult_female/gesture.json" for reference
+  poiConfig: Object  // see "3d-assets/animations/adult_female/poi.json" for reference
+}
+ ```
+ */
+async function loadAssets(
+  scene,
+  {
+    modelUrl,
+    animStandIdleUrl,
+    animLipSyncUrl,
+    animGestureUrl,
+    animEmoteUrl,
+    animFaceIdleUrl,
+    animBlinkUrl,
+    animPointOfInterestUrl,
+    gestureConfigUrl,
+    pointOfInterestConfigUrl,
+  },
+) {
+  // Load character model
+  const characterAsset = await BABYLON.SceneLoader.LoadAssetContainerAsync(modelUrl);
+  const characterMesh = characterAsset.meshes[0];
+  const bindPoseOffset = characterAsset.animationGroups[0];
+
+  // Make the offset pose additive
+  if (bindPoseOffset) {
+    BABYLON.AnimationGroup.MakeAnimationAdditive(bindPoseOffset);
+  }
+
+  characterAsset.addAllToScene();
+
+  const childMeshes = characterMesh.getDescendants(false);
+
+  const animationLoadingPromises = [
+    loadAnimation(scene, childMeshes, animStandIdleUrl, 'idleClips'),
+    loadAnimation(scene, childMeshes, animLipSyncUrl, 'lipSyncClips'),
+    loadAnimation(scene, childMeshes, animGestureUrl, 'gestureClips'),
+    loadAnimation(scene, childMeshes, animEmoteUrl, 'emoteClips'),
+    loadAnimation(scene, childMeshes, animFaceIdleUrl, 'faceClips'),
+    loadAnimation(scene, childMeshes, animBlinkUrl, 'blinkClips'),
+    loadAnimation(scene, childMeshes, animPointOfInterestUrl, 'poiClips'),
+  ];
+
+  const animLoadingResults = await Promise.all(animationLoadingPromises);
+
+  const animClips = {};
+  animLoadingResults.forEach((result) => {
+    animClips[result.clipGroupId] = result.clips;
+  });
+
+  // Load the gesture config file. This file contains options for splitting up
+  // each animation in gestures.glb into 3 sub-animations and initializing them
+  // as a QueueState animation.
+  const gestureConfig = await loadJson(gestureConfigUrl);
```

Comment:
I don't love this pattern of loading json data. This seems like something that should be baked in at build-time to cut down on network calls/etc. Not sure it is worth addressing right now though

Reply:
Regarding inclusion of generated files, committing generated API docs definitely feels like a bad idea to bad idea to me as well and something I'd love to see changed. That said, changing it would require coming up with a different solution to how we host API docs using Github Pages. I don't know our options there very well. (On Ambit, our generated doc tool automatically creates a dedicated branch for the docs when we "publish" them and that branch is served by Github Pages.)

As far as committing the generated webpack bundle output, that also feels a bit akward. But I feel strongly we should give customers _some_ easy way to quickly grab the right bundle .js file without having to download and build the whole repo. What are our options there? Releases? Force them to download the NPM package and go diving into the "node_modules" folder? Something else?

Reply:
Regarding the pattern I'm using for loading JSON data. Runtime loading of JSON is required because the correct JSON file(s) to load isn't necessarily known at build time. This function is used for loading things like "gesture" and "poi" (point-of-interest) config files which are different for each character (and for custom characters.) So if a customer wants to dynamically choose which character to instantiate at runtime, these files need to be loaded at runtime. Are you OK with me leaving this as-is?

Reply:
I don't have a lot of experience with github pages, but I know what you describe is a common pattern. It seems we should aim towards that. do we have a need for the docs to be checked in or updated right now until we get that addressed? My point being, if we know we want to do that(and I think we do) let's plan for it and stop checking in docs.

As for generated dist folder. When was the last time you github cloned a repo to get a generated file? Isn't `npm install babylon` the common route anyway? webdevs knows how to do that.


as for the json stuff. Yes, that is fine. I'm concerned about a potential pipelining problem on load, where the app needs to download the config, which downloads the gltf, which downloads the animations or whatever it is. It seems there is an opportunity to better bundle these things to optimize for load times. certainly not something we need to do right now, just worried about how it might be harder down the line.

Reply:
I like the suggestion of removing the generated docs for now. 

Regarding webdevs using npm install, you're probably right. Although in my projects I tend to link in libraries from from CDNs rather than compile those libraries into my code. I do this both for ease of development and also because it's more likely a new visitor to my site will already have a cached version of the CDN hosted lib, thus improving download times. But perhaps my practice is unusual given modern webdev practices. I'm happy to remove the compiled bundles from the repo, but will do that as a future PR since it may require some minor changes to our developer workflow (not sure.)

I agree that there's probably opportunity for optimization via bundling of JSON, 3D mesh, and animation files but will consider that out of scope for now.

Reply:
Starting in the last coupld of years, the major browsers do this thing called 'Double-key'd HTTP caching', meaning that browser caching is key'd not only by the URL of the content, but also by the origin. This is for privacy reasons, but means that if you load two websites that both use the same version of babylon.js hosted on the same CDN(which itself will be a rare occurrence), the browser is still going to download and cache it twice: https://developers.google.com/web/updates/2020/10/http-cache-partitioning

Reply:
That's great insight!